### DISCUSSION TITLE
How to add a where filter for K6-Contrib S3-images fields?

### DISCUSSION THREAD

**Discussion by PhilBookst** [keystonejs/keystone#7814] [Questions]
Hey I'm using the [K6-Contrib S3-images package](https://github.com/keystonejs-contrib/k6-contrib/tree/main/packages/fields-s3images) for my Users list and when querying for a list of users, I'd like to filter out which users have an image set but the graphql playground doesn't show any filters for the image.

This is my schema: 
```ts
export const User = list({
  fields: {
    firstName: text(),
    lastName: text(),
    image: s3Images({
      s3Config,
    }),
    username: text({
      validation: {
        isRequired: true,
      },
      isIndexed: "unique",
    }),
  },
});
```

When trying to do something like this, the playground complains. It doesn't even show image in the list of possible filters

```graphql
query Users {
  users(where: { image: ... }) {
    username
  }
}
```

I'm pretty new to graphql so I don't have much experience with it. How can I add the necessary `where` filter on the list? Do I need to extend my schema for it and if so, how?

**Comment by gautamsi** [comment]
you should have posted this question in https://github.com/keystonejs-contrib/k6-contrib

anyways, the answer is that this field does not implement any filters. this field was created from original file/image field code in `core`. 

you will not be able to filter. also this field needs to be updated to keep up with recent changes in the core file/image field.

**Reply by PhilBookst** [reply]
Yes you're right, I should have posted it in the k6-contrib repo - sorry about that.

So there's no way of knowing if an image is set on a user? Do you know any workaround on how to check if it's set? 

Thanks for your help!

**Reply by gautamsi** [reply]
for workaround you can use underlying prisma api
